lib/active_merchant/billing/gateways/flex_charge.rb

```diff
@@ -17,38 +17,61 @@ class FlexChargeGateway < Gateway
         sync: 'outcome',
         refund: 'orders/%s/refund',
         store: 'tokenize',
-        inquire: 'orders/%s'
+        inquire: 'orders/%s',
+        capture: 'capture'
       }
 
-      SUCCESS_MESSAGES = %w(APPROVED CHALLENGE SUBMITTED SUCCESS PROCESSING).freeze
+      SUCCESS_MESSAGES = %w(APPROVED CHALLENGE SUBMITTED SUCCESS PROCESSING CAPTUREREQUIRED).freeze
 
       def initialize(options = {})
         requires!(options, :app_key, :app_secret, :site_id, :mid)
         super
       end
 
       def purchase(money, credit_card, options = {})
+        options[:transactionType] = 'Purchase'
```

Comment:
Observation: When you call the `authorize` method, `options[:transactionType]` it is `'Authorization'` but then it is called within the `authorize` a `purchase` and in this line the `options[:transactionType]` field is overwritten

```suggestion
        options[:transactionType] ||= 'Purchase'
```